File: sdk/go/anyvali.go
// Package anyvali provides a native Go validation library with portable schema interchange.
package anyvali

import "math"

// String creates a new StringSchema.
func String() *StringSchema { return newStringSchema() }

// Number creates a new Float64Schema (number = float64).
func Number() *Float64Schema { return newFloat64Schema("number") }

// Float64 creates a new Float64Schema.
func Float64() *Float64Schema { return newFloat64Schema("float64") }

// Float32 creates a new Float32Schema with float32 range constraints.
func Float32() *Float32Schema { return newFloat32Schema() }

// Int creates a new Int64Schema (int = int64).
func Int() *IntSchema { return newIntSchema("int", math.MinInt64, math.MaxInt64) }

// Int8 creates a new IntSchema for int8 range.
func Int8() *IntSchema { return newIntSchema("int8", math.MinInt8, math.MaxInt8) }

// Int16 creates a new IntSchema for int16 range.
func Int16() *IntSchema { return newIntSchema("int16", math.MinInt16, math.MaxInt16) }

// Int32 creates a new IntSchema for int32 range.
func Int32() *IntSchema { return newIntSchema("int32", math.MinInt32, math.MaxInt32) }

// Int64 creates a new IntSchema for int64 range.
func Int64() *IntSchema { return newIntSchema("int64", math.MinInt64, math.MaxInt64) }

// Uint8 creates a new IntSchema for uint8 range.
func Uint8() *IntSchema { return newUintSchema("uint8", 0, math.MaxUint8) }

// Uint16 creates a new IntSchema for uint16 range.
func Uint16() *IntSchema { return newUintSchema("uint16", 0, math.MaxUint16) }

// Uint32 creates a new IntSchema for uint32 range.
func Uint32() *IntSchema { return newUintSchema("uint32", 0, math.MaxUint32) }

// Uint64 creates a new IntSchema for uint64 range.
func Uint64() *IntSchema { return newUintSchema("uint64", 0, math.MaxUint64) }

// Bool creates a new BoolSchema.
func Bool() *BoolSchema { return newBoolSchema() }

// Null creates a new NullSchema.
func Null() *NullSchema { return newNullSchema() }

// Any creates a new AnySchema.
func Any() *AnySchema { return newAnySchema() }

// Unknown creates a new UnknownSchema.
func Unknown() *UnknownSchema { return newUnknownSchema() }

// Never creates a new NeverSchema.
func Never() *NeverSchema { return newNeverSchema() }

// Literal creates a new LiteralSchema for the given value.
func Literal(v any) *LiteralSchema { return newLiteralSchema(v) }

// Enum creates a new EnumSchema for the given values.
func Enum(values ...any) *EnumSchema { return newEnumSchema(values) }

// Array creates a new ArraySchema with the given item schema.
func Array(item Schema) *ArraySchema { return newArraySchema(item) }

// Tuple creates a new TupleSchema with the given item schemas.
func Tuple(items ...Schema) *TupleSchema { return newTupleSchema(items) }

// Object creates a new ObjectSchema with the given properties.
func Object(props map[string]Schema) *ObjectSchema { return newObjectSchema(props) }

// Record creates a new RecordSchema with the given value schema.
func Record(value Schema) *RecordSchema { return newRecordSchema(value) }

// Union creates a new UnionSchema.
func Union(schemas ...Schema) *UnionSchema { return newUnionSchema(schemas) }

// Intersection creates a new IntersectionSchema.
func Intersection(schemas ...Schema) *IntersectionSchema { return newIntersectionSchema(schemas) }

// Optional wraps a schema to make it optional.
func Optional(s Schema) *OptionalSchema { return newOptionalSchema(s) }

// Nullable wraps a schema to allow null values.
func Nullable(s Schema) *NullableSchema { return newNullableSchema(s) }

File: sdk/csharp/src/AnyVali/AnyVali.cs
using AnyVali.Schemas;

namespace AnyVali;

/// <summary>
/// Static factory for creating AnyVali schemas.
/// </summary>
public static class V
{
    public static StringSchema String() => new();
    public static NumberSchema Number() => new();
    public static Float32Schema Float32() => new();
    public static Float64Schema Float64() => new();
    public static IntSchema Int() => new();
    public static Int8Schema Int8() => new();
    public static Int16Schema Int16() => new();
    public static Int32Schema Int32() => new();
    public static Int64Schema Int64() => new();
    public static Uint8Schema Uint8() => new();
    public static Uint16Schema Uint16() => new();
    public static Uint32Schema Uint32() => new();
    public static Uint64Schema Uint64() => new();
    public static BoolSchema Bool() => new();
    public static NullSchema Null() => new();
    public static AnySchema Any() => new();
    public static UnknownSchema Unknown() => new();
    public static NeverSchema Never() => new();
    public static LiteralSchema Literal(object? value) => new(value);
    public static EnumSchema Enum(params object[] values) => new(values);

    public static ArraySchema Array(Schema items) => new(items);
    public static TupleSchema Tuple(params Schema[] items) => new(items);

    public static ObjectSchema Object(
        Dictionary<string, Schema> shape,
        UnknownKeyMode unknownKeys = UnknownKeyMode.Reject) => new(shape, unknownKeys);

    public static RecordSchema Record(Schema valueSchema) => new(valueSchema);
    public static UnionSchema Union(params Schema[] variants) => new(variants);
    public static IntersectionSchema Intersection(params Schema[] schemas) => new(schemas);
    public static OptionalSchema Optional(Schema inner) => new(inner);
    public static NullableSchema Nullable(Schema inner) => new(inner);

    /// <summary>Parse input using the given schema. Throws ValidationError on failure.</summary>
    public static object? Parse(Schema schema, object? input) => schema.Parse(input);

    /// <summary>Parse input using the given schema. Returns ParseResult.</summary>
    public static ParseResult SafeParse(Schema schema, object? input) => schema.SafeParse(input);

    /// <summary>Export a schema to an AnyValiDocument.</summary>
    public static AnyValiDocument Export(Schema schema, ExportMode mode = ExportMode.Portable) =>
        Interchange.Exporter.ExportSchema(schema, mode);

    /// <summary>Import an AnyValiDocument to a live schema.</summary>
    public static Schema Import(AnyValiDocument doc) =>
        Interchange.Importer.ImportSchema(doc);
}
